packages/relay/tests/lib/eth.spec.ts

```diff
@@ -3809,6 +3827,30 @@ describe('Eth calls using MirrorNode', async function () {
       }
       expect(hasError).to.be.true;
     });
+
+    it('should return a computed hash if unable to retrieve EthereumHash from record due to contract revert', async function () {
+      restMock.onGet(accountEndpoint).reply(200, { account: accountAddress });
+    
+      const transaction = {
+        chainId: 0x12a,
+        to: accountAddress1,
+        from: accountAddress,
+        value,
+        gasPrice,
+        gasLimit: maxGasLimitHex,
+      };
+    
+      sdkClientStub.getAccountBalanceInTinyBar.returns(ethers.BigNumber.from('1000000000000000000000'));    
+      const signed = await signTransaction(transaction);
+      const id = uuid();
+    
+      restMock.onGet('network/fees').reply(200, defaultNetworkFees);
+      restMock.onGet('transactions/0.0.902-1684375868-230217103').reply(200, null);
```

Comment:
This is a good catch.  In both cleaning up the test and dealing with merge conflicts I removed a necessary piece of the test setup.  The fact that this test passed illustrates how difficult it is to test.  My testing involved stepping through the code and watching the updated condition be evaluated correctly with no logging.   Once an error is caught, the hash is generated based on the transaction buffer.  An error can be thrown for many reasons.  The test passed due to the wrong error being thrown.  The only way this test could be narrowed in scope would be to add checking for the logging response, and in this case there would be none, so the test would check that the "Transaction failed with result:" was not logged.  Perhaps a small refactor for testability should be considered?

Reply:
Per the issue you're fixing the right test would be to confirm as you've done that when the consensus node returns a contract revert that the function doesn't fail when getting the details from the mirror node.

You're right on narrowing the test conditions, it shows that `extractContractRevertReason()` is probably not best situated in `eth.ts`, probably should be in  another class of util nature that would allow for testing that function specifically.
Maybe move it to `MirrorNodeClient.ts` as a `getContractRevertReasonFromTransaction`, then you can have 2 test verify a 200 and a 404 case in the `mirrorNodeClient.spec.ts`.

Reply:
Agreed, and updated.